src/AwsServiceProvider.php

```diff
@@ -37,30 +44,37 @@ class AwsServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        $this->app['aws'] = $this->app->share(function ($app) {
-            // Retrieve the config
-            $config = $app['config']['aws'] ?: $app['config']['aws::config'];
-            if (isset($config['config_file'])) {
-                $config = $config['config_file'];
-            }
+        $this->app->singleton('aws', function ($app) {
 
             // Instantiate the AWS service builder
-            $aws = Aws::factory($config);
+            $aws = Aws::factory($app['config']->get('aws'));
 
             // Attach an event listener that will append the Laravel and module version numbers to the user agent string
             $aws->getEventDispatcher()->addListener('service_builder.create_client', function (Event $event) {
-                $clientConfig = $event['client']->getConfig();
-                $commandParams = $clientConfig->get(Client::COMMAND_PARAMS) ?: array();
-                $userAgentSuffix = 'Laravel/' . Application::VERSION . ' L4MOD/' . AwsServiceProvider::VERSION;
-                $clientConfig->set(Client::COMMAND_PARAMS, array_merge_recursive($commandParams, array(
+                $clientConfig       = $event['client']->getConfig();
+                $commandParams      = $clientConfig->get(Client::COMMAND_PARAMS) ?: [];
+                $userAgentSuffix    = 'Laravel/' . Application::VERSION . ' L5MOD/' . AwsServiceProvider::VERSION;
+
+                $clientConfig->set(Client::COMMAND_PARAMS, array_merge_recursive($commandParams, [
                     UserAgentListener::OPTION => $userAgentSuffix,
-                )));
+                ]));
             });
 
             return $aws;
         });
     }
 
+    public function prepareConfig()
```

Comment:
This doesn't need it's own method. Just put it in the boot method. It's also convention to have the boot method above the register method.